memphis/memphis.go

```diff
@@ -28,17 +31,77 @@ type MemphisOutput struct {
 	FailedMessages []MemphisMsgWithError `json:"failed_messages"`
 }
 
-// EventHandlerFunction gets the message payload as []byte, message headers as map[string]string and inputs as map[string]string and should return the modified payload and headers.
+// HandlerType functions get the message payload as []byte (or any), message headers as map[string]string and inputs as map[string]string and should return the modified payload and headers.
 // error should be returned if the message should be considered failed and go into the dead-letter station.
 // if all returned values are nil the message will be filtered out of the station.
-type EventHandlerFunction func([]byte, map[string]string, map[string]string) ([]byte, map[string]string, error)
+type HandlerType func(any, map[string]string, map[string]string) (any, map[string]string, error)
+
+type HandlerOption func(*HandlerOptions) error
+
+type HandlerOptions struct {
+	Handler           HandlerType
+	UserObject        any
+}
+
+func ObjectOption(schema any) HandlerOption {
+	return func(handlerOptions *HandlerOptions) error {
+		handlerOptions.UserObject = schema
+		return nil
+	}
+}
+
+func UnmarshalIntoStruct(data []byte, userStruct any) error {
+	// Unmarshal JSON data into the struct
+	err := json.Unmarshal(data, userStruct)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func checkValidStruct(userStruct any) error {
+	// Check if the provided variable is a pointer to a struct
+	valueOf := reflect.ValueOf(userStruct)
+	if valueOf.Kind() != reflect.Ptr || valueOf.Elem().Kind() != reflect.Struct {
+		return fmt.Errorf("input parameter must be a pointer to a struct")
+	}
+
+	return nil
+}
 
 // This function creates a Memphis function and processes events with the passed-in eventHandler function.
-// eventHandlerFunction gets the message payload as []byte, message headers as map[string]string and inputs as map[string]string and should return the modified payload and headers.
+// eventHandler gets the message payload as []byte or as the user specified type, 
+// message headers as map[string]string and inputs as map[string]string and should return the modified payload and headers.
+// The modified payload type will either be the user type, or []byte depending on user requirements.
 // error should be returned if the message should be considered failed and go into the dead-letter station.
 // if all returned values are nil the message will be filtered out from the station.
-func CreateFunction(eventHandler EventHandlerFunction) {
+func CreateFunction(eventHandler HandlerType, options ...HandlerOption) {
 	LambdaHandler := func(ctx context.Context, event *MemphisEvent) (*MemphisOutput, error) {
+		params := HandlerOptions{
+			Handler:           eventHandler,
+			UserObject:        nil,
+		}
+
+		for _, option := range options {
```

Comment:
move to before LambdaHandler and pass it inside

Reply:
The lambda handler is given to the lambda.Start, and must follow their  signature:
https://docs.aws.amazon.com/lambda/latest/dg/golang-handler.html

> func HandleRequest(ctx context.Context, event *MyEvent) (*string, error): This is the signature of your Lambda handler. It's the entry point for your Lambda function and contains the logic that is executed when your function is invoked. In addition, the parameters included denote the following:

In order for Lambda to work, this must maintain the same. I cannot pass these in the LambdaHandler, they muse be used like this.